File: SHPC4001_fortrancode/bouncing.f90
program bouncing_balls
implicit none

! Use the Euler method to compute the trajectory of a bouncing
! ball assuming perfect reflection at the surface x = 0.
! Use SI units (meters and seconds)

integer :: steps
real :: x, v, g, t, dt

x = 1.0	! initial height of the ball
v = 0.0	! initial velocity of the ball
g = 9.8	! gravitational acceleration
t = 0.0	! initial time
dt = 0.01	! size of time step

open(10,file='bounce.dat')   ! open data file

do steps = 1, 300    ! loop for 300 timesteps

  t = t + dt
  x = x + v*dt
  v = v - g*dt

  if(x.lt.0) then		! reflect the motion of the ball
    x = -x			! when it hits the surface x=0
    v = -v
  endif
  write(10,*) t, x, v	! write out data at each time step

enddo

end

File: SHPC4001_fortrancode/bouncing.c
/*
 *  bouncing.c
 *  
 *  A c program calculating the trajectory of a bouncing ball.
 *  Uses the Euler methof to compute the trajectory of a bouncing 
 *  ball assuming perfect reflection at the surface x = 0
 *  USE SI units (meters and seconds)
 *  Created by Filip Welna on Wed Aug 14 2002.
 *  Copyright (c) 2002 UWA physics All rights reserved.
 *
 */

#include <stdio.h>

int main(void)
{

    int steps;
    float x,v,t,g,dt;
    FILE* output_file;
    
    x = 1.0;
    v = 0.0;
    g = 9.8;
    dt = 0.01;
    t = 0.0;
    
    output_file = fopen("bounce.dat","w");
    
    for(steps=1;steps<=300;steps++)
    { 
        t += dt;     /*shorthand equivalent to t = t+dt */ 
        x += v*dt;
        v -= g*dt;
        if(x < 0)
        {
            x = -x;
            v = -v;
        }
        
        fprintf(output_file,"%f %.7f %f \n",t,x,v); 
       
    }
     fclose(output_file);
}

